src/definitions.jl

```diff
@@ -287,9 +289,12 @@ for f in (:brfft, :irfft)
 end
 
 for f in (:brfft, :irfft)
+    pf = Symbol("plan_", f)
```

Comment:
It was unclear to me why the plans for backwards and inverse real FFTs did not automatically convert to complex floats (instead they throw an error given a real array), so I copied over the analogous logic for regular fft's. (Needed for rrule's to work.)